modules/ec2_instances/templates/user_data.sh

```diff
@@ -87,7 +87,7 @@ EOF
 
 # Mount the EFS
 sudo mkdir /opt/data
-sudo mount -t nfs4 -o nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2 ${efs_id}.efs.ap-southeast-2.amazonaws.com:/ /opt/data/
+sudo mount -t nfs4 -o nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2 ${efs_id}.efs.$region.amazonaws.com:/ /opt/data/
```

Comment:
Injected variables should be in the form ${region} and will need to be passed to the file in the modules/ec2_instances/main.tf data "templeate_file" "userdata"